text/0000-lwc-dynamic-with-attrs.md

```diff
@@ -0,0 +1,141 @@
+---
+title: LWC Dynamic with Attrs/Props/Event listeners
+status: DRAFTED
+created_at: 2021-10-01
+updated_at: 2021-10-01
+pr: (leave this empty until the PR is created)
+---
+
+# LWC Dynamic with Attrs/Props/Event listeners
+
+## Summary
+
+This proposal augments `lwc:dynamic` to not just accept the constructor, but also
+the properties that are passed to the component. This allows them to be changed dynamically.
+Support for attributes/event listeners will be added later if required.
+
+## Terminology
+*component config*: properties/constructor listeners on a component. Will be amended later to include attributes/event listeners.
+
+## Basic example
+
+```html
+// x/app.html
+<template>
+    <x-cmp lwc:dynamic={cmpConfig}></x-cmp>
+</template>
+```
+
+```js
+// x/app.js
+import { LightningElement } from 'lwc';
+
+export default class App extends LightningElement {
+    cmpConfig;
+    async loadCtor() {
+        const constructor = await import("x/child");
+
+        const props = { name: 'John Foo' };
+
+        this.cmpConfig = { constructor, props };
```

Comment:
@abdulsattar In [the PR](https://github.com/salesforce/lwc/pull/2991) we have this as `ctor` rather than `constructor`, but it looks like this RFC hasn't been updated.